src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    static final int DEFAULT_CAPACITY = 10;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        this.elements = (T[]) new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
+
+    private void resize() {
+        T[] resizedArray = (T[]) new Object[(elements.length + (elements.length >> 1))];
+        System.arraycopy(elements, 0, resizedArray, 0, size);
+        elements = resizedArray;
+    }
 
     @Override
     public void add(T value) {
-
+        if (size == elements.length) {
+            resize();
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new ArrayIndexOutOfBoundsException(index + " is out of bounds");
+        } else if (size == elements.length) {
+            resize();
+        } else {
+            System.arraycopy(elements, index, elements, index + 1, size - index);
+            if (index == size) {
+                add(value);
+            } else {
+                set(value, index);
+                size++;
+            }
+        }
 
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= size || index < 0) {
+            throw new ArrayIndexOutOfBoundsException(index + " is out of bounds");
+        } else {
```

Comment:
you don't need else after throw new Exception

Reply:
removed